File: backend/src/main/java/com/globant/youtube_clone/dto/VideoDTO.java
package com.globant.youtube_clone.dto;

import com.globant.youtube_clone.model.VideoStatus;

import java.time.LocalDate;
import java.util.Set;

public record VideoDTO (
        String id,
        String title,
        String description,
        String userId,
        Set<String>tags,
        String videoUrl,
        VideoStatus videoStatus,
        String thumbnailUrl,
        Integer likes,
        Integer dislikes,
        Integer views,
        LocalDate uploadedAt
) {
}

File: frontend/youtuble-clone-ui/src/app/video-dto.ts
export interface VideoDTO {
  id: string;
  title: string;
  description: string;
  userId: string;
  tags: Array<string>;
  videoUrl: string;
  videoStatus: string;
  thumbnailUrl: string;
  likes: number;
  dislikes: number;
  views: number;
  uploadedAt: string;
}
